servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/DefaultLoadBalancer.java

```diff
@@ -179,6 +154,34 @@ private void subscribeToEvents(boolean resubscribe) {
         }
     }
 
+    // This method is called eagerly, meaning the completable will be immediately subscribed to,
+    // so we don't need to do any Completable.defer business.
+    private Completable doClose(final boolean graceful) {
+        CompletableSource.Processor processor = Processors.newCompletableProcessor();
+        sequentialExecutor.execute(() -> {
+            if (!isClosed) {
+                discoveryCancellable.cancel();
```

Comment:
lets put other code in try/finally since this may call externally and throw.

Reply:
You're right. I think I've done what you've suggested but it's worth a double check.

I also feel this is messier than it needs to be, but otoh the io.st.concurrent.api.Executor is a bit thicker of an API than I think we want. Any ideas appreciated.